### ISSUE TITLE
Bridging 3 or more endpoints using SqlTransport can cause the ReplyToAddress to be incorrect and prevent Replying to messages between bridged endpoints

### ISSUE THREAD

**Issue by awright18** [Particular/NServiceBus.MessagingBridge#369]
#### Description
When bridging endpoints using more than 2 Sql Endpoints the `NServiceBus.ReplyToAddress` header is set using the last configured endpoint's database. 

For example if you have 3 endpoints 

endpoint1,endpoint2,endpoint3 and the transports are added in that order to the BridgeConfiguration

```c#   
   //code simplified for brevity
     bridgeConfiguration.AddTransport(bridgeTransport1);
     bridgeConfiguration.AddTransport(bridgetTransport2);
     bridgeConfiguration.AddTransport(bridgeTransport4);
     bridgeConfiguration.AddTransport(bridgeTransport3);      
```

and each endpoint has it's own database Db1, Db2, Db3 respectively.

Each database has a user account that only has permission's to access it. (Db1User,Db2User,Db3User).

If Endpoint1 sends a command to Endpoint2. 

Endpoint2 will see the ReplyToAdress as Endpoint1@[dbo]@[DB3] 

This is because the last transport configured was endpoint3 and its' database is DB3.

When attempting to send the reply if DB3 is hosted on the same SqlServer instance then an exception will be thrown stating 

"The server principal "Db2User" is not able to access the database "Db3" under the current security context". 

If Db3 is hosted on a separate instance, then an exception will be thrown stating that "Invalid object DB3.dbo.Endpoint1"
 
#### Expected behavior

ReplyToAddress will use the originating Database and Endpoint in the address instead of on relying on the last configured endpoint in the bridge. 

Endpoints using the sql transport hosted on separate instances or using database specific credentials can reply to messages from on and from any instance regardless of where they are hosted. 

#### Actual behavior

When 3 or more endpoints using the SqlTransport are configure in the bridge ReplyToAddresses are composed of the originating endpoint and the last configured database instead of the originating database. This causes Replies to messages to throw exceptions in cases where the database doesn't match the sender.

#### Versions

Please list the version of the relevant packages or applications in which the bug exists.        

Recreated with the following versions
- NServiceBus.MessagingBridge version 2.1
- NServiceBus.Transport.SqlServer 7.0.2

### Steps to reproduce

A repository exists at https://github.com/awright18/NSBBridgeSqlTest/tree/net472 which has 4 SQL Transports configured. This can be modified to setup the different scenarios described in the bug. 

Scenario 1: One Sql Instance with separate users 
1. Create a database named Db1
2. Create a Login named Db1User
3. Create or map the Db1User login to a user in Db1 database named Db1User
4. Assign the Db1User in Db1 db_owner permissions
5. Create a database named Db2
6. Create a Login named Db2User
7. Create or map the Db2User login to a user in Db2 database named Db3User
8. Assign the Db2User in Db2 db_owner permissions
9. Create a database named Db3
10. Create a Login named Db3User
11. Create or map the Db3User login to a user in Db3 database named Db3User
12. Assign the Db3User in Db3 db_owner permissions

11. Update the connection string for Db1 in the Bridge project to use the Db1User User Id
12. Update the persistence and transport connection strings in the Endpoint1 project to use the Db1User User Id

13. Update the connection string for Db2 in the Bridge project to use the Db2User User Id
15. Update the persistence and transport connection strings in the Endpoint2 project to use the Db2User User Id 

13. Update the connection string for Db3 in the Bridge project to use the Db3User User Id
15. Update the persistence and transport connection strings in the Endpoint3 project to use the Db2User User Id 

17. Start the Bridge app
18. Start Endpoint1 and the Endpoint2 app
19. In Endpoint1 app press 1 to send a message

Endpoint2 will throw when trying to Reply to the CreateOrder command.

### Relevant log output

```shell
Failed to send message to [Db3].[dbo].[Endpoint1] ---> Microsoft.Data.SqlClient.SqlException: Invalid object name 'Db3.dbo.Endpoint1'
```

#### Additional information
A sample repo is setup to verify this problem. Databases will likely need to be created and connection strings altered though.

**Comment by awright18** [comment]
Hey @andreasohlund , thanks for taking a look. That's a nice find! However, I forgot that I had changed one of the endpoint transports to use MSMQ instead of Sql for testing purposes. The problem occurs with 3 or more SqlTranports. I'll change the sample back to using all sql transports so you can see the actual issue.

**Comment by andreasohlund** [comment]
Ah, but in your case you have more than one SQL so that won't fly, in my case I had only one :(

**Comment by jpalac** [comment]
This will be released in 4.0.2, 3.1.1, 2.3.1
